File: elixir/scl_parser/lib/scl/parser/tokens.ex
defmodule SCL.Parser.Tokens do
  @moduledoc false

  @type token_type ::
          :identifier
          | :double_colon
          | :bool
          | :str
          | :num
          | :fl
          | :ml
          | :class
          | :list
          | :dynamic
          | :lbrace
          | :rbrace
          | :lparen
          | :rparen
          | :comma
          | :string
          | :multiline_string
          | :number
          | :float
          | :boolean
          | :comment
          | :newline
          | :eof

  @type t :: %{type: token_type(), value: any(), line: non_neg_integer(), column: non_neg_integer()}
end

File: python/scl_parser/tokens.py
from dataclasses import dataclass
from enum import Enum
from typing import Any, Optional


class TokenType(Enum):
    IDENTIFIER = "IDENTIFIER"
    DOUBLE_COLON = "::"
    BOOL = "bool"
    STR = "str"
    NUM = "num"
    FL = "fl"
    ML = "ml"
    CLASS = "class"
    LIST = "list"
    DYNAMIC = "dynamic"
    LBRACE = "{"
    RBRACE = "}"
    LPAREN = "("
    RPAREN = ")"
    COMMA = ","
    STRING = "STRING"
    MULTILINE_STRING = "MULTILINE_STRING"
    NUMBER = "NUMBER"
    FLOAT = "FLOAT"
    BOOLEAN = "BOOLEAN"
    COMMENT = "COMMENT"
    NEWLINE = "NEWLINE"
    EOF = "EOF"


@dataclass
class Token:
    type: TokenType
    value: Any
    line: int
    column: int


